packages/rocketchat-livechat/server/startup.js

```diff
@@ -11,6 +11,19 @@ Meteor.startup(() => {
 		return room.t === 'l' && extraData && extraData.token && room.v && room.v.token === extraData.token;
 	});
 
+	RocketChat.authz.addRoomAccessValidator(function(room, file) {
+		return room.t === 'l' && file && file.visitorToken && room.v && room.v.token === file.visitorToken;
+	});
+
+	RocketChat.authz.addRoomAccessValidator(function(file) {
+		if (!file || !file.rid || !file.visitorToken) {
+			return;
+		}
+
+		const room = RocketChat.models.Rooms.findOneById(file.rid);
+		return room && room.t === 'l' && file && file.visitorToken && room.v && room.v.token === file.visitorToken;
```

Comment:
you don't need to check `file && file.visitorToken` here again, you checked them before already